FilePersistence/src/version_control/merge/ChangeGraph.cpp

```diff
@@ -448,14 +446,105 @@ void ChangeGraph::removeLabelConflictsBetweenChildren(Model::NodeIdType parentId
 	}
 }
 
-void ChangeGraph::updateIncomingLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/)
+void ChangeGraph::updateListChanges(Model::NodeIdType parentId, IdToLabelMap labelMap, GenericTree* baseTree)
 {
-	Q_ASSERT(false);
-}
+	for (auto nodeId : labelMap.keys())
+	{
+		QString labelA;
+		QString labelB;
+		QString labelNone;
+		auto labels = labelMap.values(nodeId);
+		for (auto label : labels)
+		{
+			if (label.branch_.testFlag(MergeChange::BranchA))	labelA = label.label_;
+			else if (label.branch_.testFlag(MergeChange::BranchB))	labelB = label.label_;
+			else labelNone = label.label_;	//Base
+		}
 
-void ChangeGraph::updateOutgoingLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/)
-{
-	Q_ASSERT(false);
+		// Need to adjust Update Changes as well
+		auto parentNode = baseTree->find(parentId);
+		Q_ASSERT(parentNode != NULL);
+		auto childNode = baseTree->find(nodeId);
+		if (childNode && parentNode->children().contains(childNode))	// Deletion, Move Out, LabelChange, Stable
```

Comment:
Hmm, I'm not sure there is a need to look for the child node at all. It seems like we only do this for checking, but we can do this while iterating over all changes for nodeId. We should discuss this.